tailcfg/tailcfg.go

```diff
@@ -336,6 +336,10 @@ type Node struct {
 	// is not expected to speak Disco or DERP, and it must have Endpoints in
 	// order to be reachable.
 	IsWireGuardOnly bool `json:",omitempty"`
+
+	// DNSServers is the list of DNS servers that should be used when this
```

Comment:
can this be defined more? How does this interact with DNSConfig split DNS magic DNS etc? Is this just a fallback?

Cc @danderson

Reply:
It should have the same semantics as the the peer API path.

There are other issues we need to look into with the peer API path, which are entirely separate, if we address those issues, some specific semantics may change, regardless, this particular field should mimic the semantics of the peer API behavior for exit node DNS.

Reply:
This seems straightforward to me as described. Our normal exit node behavior with DNS is to forward all resolution traffic except MagicDNS to the exit node, via peerapi. For wireguard-only peers, this breaks because those peers don't have a peerapi. But we still need to forward DNS somewhere, because in the default config exit nodes prevent LAN traffic and so break local DNS. So, this field says "if this is a wireguard-only peer being used as an exit node, here's the DNS config to use to emulate peerapi forwarding behavior".

If we try to describe all the possible interactions of different DNS configs here, the comment will become obsolete almost immedaitely, so I'd prefer to not try and write down a spec for our full DNS semantics in this comment. The comment as-is seems as descriptive to me as the rest of our DNS config structs in tailcfg, which all describe a data plane mechanism without specifying 100% of every interaction with every other feature. Those decisions should live in the DNS manager code, and its tests.

Reply:
If SplitDNS works with the peer API configuration, it should work with the configuration produced as a result of consuming this field.

Reply:
This should be named appropriately  like `ExitNodeDNSServers`